src/coreclr/vm/syncblk.cpp

```diff
@@ -2541,22 +2541,22 @@ BOOL AwareLock::EnterEpilogHelper(Thread* pCurThread, INT32 timeOut)
     // the object associated with this lock.
     _ASSERTE(pCurThread->PreemptiveGCDisabled());
 
-    BOOLEAN IsContentionKeywordEnabled = ETW_TRACING_CATEGORY_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_DOTNET_Context, TRACE_LEVEL_INFORMATION, CLR_CONTENTION_KEYWORD);
+    LogContention();
+    Thread::IncrementMonitorLockContentionCount(pCurThread);
+
+    OBJECTREF obj = GetOwningObject();
+
+    BOOLEAN isContentionKeywordEnabled = IsContentionKeywordEnabled();
     LARGE_INTEGER startTicks = { {0} };
 
-    if (IsContentionKeywordEnabled)
+    if (isContentionKeywordEnabled)
     {
         QueryPerformanceCounter(&startTicks);
 
         // Fire a contention start event for a managed contention
-        FireEtwContentionStart_V1(ETW::ContentionLog::ContentionStructs::ManagedContention, GetClrInstanceId());
+        FireEtwContentionStart_V2(ETW::ContentionLog::ContentionStructs::ManagedContention, GetClrInstanceId(), OBJECTREFToObject(obj), this);
```

Comment:
The two new parameters are LockID and OwningThreadID, so the first new parameter should be `this`, and the second should be `pCurThread->GetOSThreadId64()`.

Reply:
Oh yeah you are right... I forgot this one. I was focused on the AwareLock event.

Reply:
I think it should be `m_HoldingThread` instead of `pCurThread`, because we want the thread that currently holds the lock

Reply:
Yes, you're right.